src/detect-engine.c

```diff
@@ -117,12 +117,10 @@ SCEnumCharMap det_ctx_event_table[] = {
     { "Z_STREAM_ERROR", FILE_DECODER_EVENT_Z_STREAM_ERROR },
     { "Z_BUF_ERROR", FILE_DECODER_EVENT_Z_BUF_ERROR },
     { "Z_UNKNOWN_ERROR", FILE_DECODER_EVENT_Z_UNKNOWN_ERROR },
+    { "LZMA_IO_ERROR", FILE_DECODER_EVENT_LZMA_IO_ERROR },
+    { "LZMA_HEADER_TOO_SHORT_ERROR", FILE_DECODER_EVENT_LZMA_HEADER_TOO_SHORT_ERROR },
     { "LZMA_DECODER_ERROR", FILE_DECODER_EVENT_LZMA_DECODER_ERROR },
-    { "LZMA_MEMLIMIT_ERROR", FILE_DECODER_EVENT_LZMA_MEMLIMIT_ERROR },
```

Comment:
Do we lose this mem limit error ?

Reply:
When lzma-rs exceeds the memory cap for decoding, it returns a `LzmaError`, which is mapped to `FILE_DECODER_EVENT_LZMA_DECODER_ERROR` here. The functionality it still there, but the event name has changed. Other kinds of errors are also returned as `LzmaError`, so there is no specific error type for "You exceeded your memory quota or I could not allocate memory for you" (which is what `SZ_ERROR_MEM` semantically mapped to).

Reply:
Ok, would it be possible/simple for lzma-rs to do this ?

Reply:
That would be a major bump for them to change the public API for their `Error` type to add a specific variant.

It would be trivial though to check the error message. When lzma-rs produces this kind of error the error message is obvious. They have a test that does this [here](https://github.com/gendx/lzma-rs/blob/17bb25f74736f4e15ca2d2d7a7044b9ee71ec972/tests/lzma.rs#L332) and it would be trivial to check the error message in our lzma.rs and map it to a different local error type.